includes/WikiManager.php

```diff
@@ -377,8 +377,19 @@ private function logEntry( string $log, string $action, string $actor, string $r
 	}
 
 	private function recacheJson( $wiki = null ) {
-		$cWJ = new CreateWikiJson( $wiki ?? $this->config->get( 'CreateWikiGlobalWiki' ), $this->hookRunner );
-		$cWJ->resetDatabaseList();
-		$cWJ->update();
+        $cWJ = new CreateWikiJson($wiki ?? $this->config->get('CreateWikiGlobalWiki'), $this->hookRunner);
+        $cWJ->resetDatabaseList();
+        $cWJ->update();
+    }
+    public function getUserRequestCount(string $userID) {
+        $requestCount = $this->cwdb->selectRowCount(
```

Comment:
This DB query must take into account suppressed wiki requests, else this leaks how many suppressed wiki requests have been authored by a given user. You have to check for suppressed wiki requests and if the wiki creator has the relevant permission to see the wiki request. This system has been reworked recently, check out [T12010](https://issue-tracker.miraheze.org/T12010) for the new system.